### PR TITLE
Updated platform (api, token, project) params to be optional

### PR DESCRIPTION
**Pull request by garethahealy**
#### What is this PR About?
jenkins-client-plugin defaults the project and cluster if not provided, so these params should not be mandatory

#### How do we test this?
Run TESTING.md

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

File: vars/buildAndTag.groovy [hunk 1]
```diff
@@ -5,12 +5,14 @@ class BuildAndTagInput implements Serializable {
...
-    String clusterAPI              = ''
-    String clusterToken            = ''
     String buildProjectName        = ''
     String fromFilePath            = ''
     String tagDestinationTLSVerify = 'false'
     String tagSourceTLSVerify      = 'false'
+
+    //Optional - Platform
+    String clusterAPI              = ""
+    String clusterToken            = ""
```
Comment (pabrahamsson) [inline vars/buildAndTag.groovy]: Should `buildProjectName` also be under optional?

File: vars/configMap.groovy [hunk 2]
```diff
@@ -14,7 +14,6 @@ def call(Map input) {
 }
 
 def call(ConfigMapInput input) {
-    assert input.projectName?.trim() : "Param projectName should be defined."
     assert input.configMapName?.trim()  : "Param configMapName should be defined."
 
     echo "Read ConfigMap: ${input.projectName}/${input.configMapName}"
```
Comment (pabrahamsson) [inline vars/configMap.groovy]: Can we either remove this line or change it such that we don't echo `Read ConfigMap: /<configMap>` when `projectName is empty?

Comment (garethahealy) [reply vars/configMap.groovy]: Moved it into the DSL block, so it always prints out the project which the DSL has selected

**Comment by garethahealy** [comment]
@pabrahamsson ; pushed requested changes.

**Pull request merged**